src/main/java/stream/flarebot/flarebot/permissions/PerGuildPermissions.java

```diff
@@ -2,29 +2,30 @@
 
 import net.dv8tion.jda.core.Permission;
 import net.dv8tion.jda.core.entities.Member;
+import net.dv8tion.jda.core.entities.Role;
 import stream.flarebot.flarebot.FlareBot;
 import stream.flarebot.flarebot.commands.Command;
 
+import java.util.ArrayList;
 import java.util.Collection;
 import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
 import java.util.Map;
 import java.util.concurrent.ConcurrentHashMap;
 
 public class PerGuildPermissions {
 
     private final ConcurrentHashMap<String, Group> groups = new ConcurrentHashMap<>();
-    private final ConcurrentHashMap<String, User> users = new ConcurrentHashMap<>();
-    protected String id;
+    private final ConcurrentHashMap<Member, User> users = new ConcurrentHashMap<>();
```

Comment:
Using a String or Long for the User ID would be best. Don't really want to serialise a member for every group for every guild